spec.html

```diff
@@ -1024,6 +1024,7 @@ <h1>Mathematical Operations</h1>
       <emu-note>
         <p><emu-eqn>floor(_x_) = _x_ - (_x_ modulo 1)</emu-eqn>.</p>
       </emu-note>
+      <p>An <dfn id="interval">interval</dfn> from lower bound _a_ to upper bound _b_ is a possibly-infinite, possibly-empty set of numeric values of the same numeric type. A numeric value _x_ is included in the interval if it satisfies both its <em>lower bound condition</em> and its <em>upper bound condition</em>. If an interval's lower bound is described as inclusive, its lower bound condition is _a_ &le; _x_; otherwise it is _a_ &lt; _x_. If an interval's upper bound is described as inclusive, its upper bound condition is _x_ &le; _b_; otherwise it is _x_ &lt; _b_. For example, the interval from 1 (inclusive) to 2 (exclusive) consists of all mathematical values between 1 and 2, including 1 and not including 2. An <dfn id="inclusive-interval">inclusive interval</dfn> is an interval where both bounds are inclusive. It is an editorial error if any bound of an interval is not described as either inclusive or exclusive. For the purpose of defining intervals, *-0*<sub>𝔽</sub> &lt; <!-- this comment here to avoid ecmarkup complaining about comparison to 0 --> *+0*<sub>𝔽</sub>, so, for example, an inclusive interval with a lower bound of *+0*<sub>𝔽</sub> includes *+0*<sub>𝔽</sub> but not *-0*<sub>𝔽</sub>. *NaN* is never included in an interval.</p>
```

Comment:
> `If an interval's lower bound is described as inclusive, its lower bound condition is _a_ &le; _x_; otherwise [...]`

I would replace "otherwise" with "if it is described as exclusive". Seems weird not to mention "exclusive" in this part of the definition.

Reply:
How is this? https://github.com/tc39/ecma262/pull/2848/commits/19bd4f621f27b83e73efd92e5598c0ea45a9bb06

Reply:
I still would much prefer the wording I suggested. As mentioned previously, I very specifically do not want to give a definition in the case that one of the bounds is not described as being either inclusive or exclusive.

Reply:
@bakkot I think you've misread it. Can you look again?

Reply:
> If an interval's lower bound is described as inclusive, its lower bound condition is _a_ &le; _x_; otherwise, it's lower bound is exclusive and its lower bound condition is _a_ &lt; _x_.

This says to me that if the bound is not described as being inclusive then it is assumed to be exclusive. That is the specific thing I want to avoid doing.

Reply:
I would certainly prefer to avoid implicit boundaries, and to have inclusiveness and exclusiveness explicitly indicated.

Reply:
@ljharb It already always is explicit. This is about whether the notation would even be defined for such an editorial error.

@bakkot We already explicitly describe that scenario as an editorial error. I don't think we also need to go out of our way to avoid the (true!) implication that the two are mutually exclusive. We're going to sacrifice clarity going through these gymnastics.

Reply:
I still don't like your suggested wording. Saying "if inclusive" and then "if exclusive" feels like it's intentionally trying to permit a case where a bound is both inclusive and exclusive.

I really don't think it's necessary to be trying to prevent this error by construction. How about we just add a lint rule for the usage?

Reply:
I strongly prefer that we avoid defining it entirely. I am open to another wording which accomplishes this if you have one to suggest, but I am against any wording which gives picks one of the two to be the default when neither "inclusive" nor "exclusive" is used, even if we scrupulously avoid ever being in that scenario.

Reply:
If you want to be more explicit that an interval can't have one bound be both inclusive and exclusive - which really doesn't feel like something we need to tell our readers, but whatever - you can add a sentence before this one which says "Either condition may be either inclusive or exclusive", or something to that effect.

Reply:
@bakkot This wording doesn't have a default. I feel like we're looking at different things.

Reply:
> If an interval's lower bound is described as inclusive, its lower bound condition is a ≤ x; otherwise, it's lower bound is exclusive

How is that not a default? There is "described as inclusive" and there is "otherwise".

I want the cases to be "described as inclusive" and "described as exclusive".

Reply:
Otherwise, it is described as exclusive. The "otherwise" is there to make the two cases mutually exclusive, not to provide some default in the absence of a descriptor. We literally say

> It is an editorial error if any bound of an interval is not described as either inclusive or exclusive.

to make that impossible! I just can't fathom how that's not sufficient for you.

Reply:
To say "it is an editorial error to write X" is different from _declining to give a definition to X_, and the second thing is the thing I want us to do.